sr_fingertip_visualization/src/sr_fingertip_visualization/finger_widgets_visual.py

```diff
@@ -0,0 +1,328 @@
+        self._fingers = ['ff', 'mf', 'rf', 'lf', 'th']
+        self._data = dict()
+        self._finger = finger
+        self._side = side
+        self._timer = QTimer()
+        self._subscriber = None
+
+        self.setTitle(finger)
+        self.setCheckable(True)
+        self.setChecked(False)
+        self.setSizePolicy(1, 1)
+        self.clicked.connect(self.refresh)
+
+        layout = QVBoxLayout()
+        layout.addWidget(self._tactile_point_widget, alignment=Qt.AlignCenter)
+        self._tactile_data_callback(rospy.wait_for_message('/{}/tactile'.format(self._side), BiotacAll))
+        self.setLayout(layout)
+
+    def refresh(self, state):
+        if state:
+            self.start_timer_and_subscriber()
+        elif not state:
+            self.stop_timer_and_subscriber()
+
+    def start_timer_and_subscriber(self):
+        self._subscriber = rospy.Subscriber('/{}/tactile'.format(self._side), BiotacAll, self._tactile_data_callback)
+        self._timer.timeout.connect(self.timerEvent)
+        self._timer.start(10)
+
+    def stop_timer_and_subscriber(self):
+        self._timer.stop()
+        if self._subscriber:
+            self._subscriber.unregister()
+
+    def _tactile_data_callback(self, data):
+        for i, finger in enumerate(self._fingers):
+            if finger == self._finger:
+                for data_field in self._tactile_point_widget.get_data_fields():
+                    if data_field == "pac0":
+                        self._data[data_field] = data.tactiles[i].pac0
+                    elif data_field == "pac1":
+                        self._data[data_field] = data.tactiles[i].pac1
+                    elif data_field == "pdc":
+                        self._data[data_field] = data.tactiles[i].pdc
+                    elif data_field == "tac":
+                        self._data[data_field] = data.tactiles[i].tac
+                    elif data_field == "tdc":
+                        self._data[data_field] = data.tactiles[i].tdc
+
+    def timerEvent(self):
+        self._tactile_point_widget.update_data(self._data)
+
+
+class FingerWidgetVisualBiotacSPPlus(QGroupBox):
+    def __init__(self, side, finger, parent):
+        super().__init__(parent=parent)
+        self._fingers = ['ff', 'mf', 'rf', 'lf', 'th']
+        self._version = 'v2'
+        self._data = dict()
+        self._finger = finger
+        self._side = side
+        self._timer = QTimer()
+        self._data_bar = BiotacSPPlusInfo()
+        self._datatype_to_display = 'electrodes'
+        self._subscriber = None
+
+        self.setTitle(finger)
+        self.setCheckable(True)
+        self.setChecked(False)
+        self.setSizePolicy(1, 1)
+        self.clicked.connect(self.refresh)
+
+        self._coordinates = dict()
+        self._coordinates['v1'] = dict()
+        self._coordinates['v1']['sensing'] = dict()
+        self._coordinates['v1']['sensing']['x'] = [6.45, 3.65, 3.65, 6.45, 3.65, 6.45, 0.00, 1.95, -1.95, 0.00,
```

Comment:
Why not having all these values in a json or yaml file properly formatted? That would make the code more readable and improve maintainability